dmn/flight-has-capacity.dmn

```diff
@@ -0,0 +1,133 @@
+    <itemComponent id="_1A672F6E-8172-4B6E-88FD-53DE6AD52D68" name="Flight Number" isCollection="false" typeLanguage="https://www.omg.org/spec/DMN/20230324/FEEL/">
+      <typeRef>string</typeRef>
+    </itemComponent>
+  </itemDefinition>
+  <itemDefinition id="_C041D4AC-CB51-40D0-8879-DB51FDA5EB1D" name="tFilght" isCollection="false" typeLanguage="https://www.omg.org/spec/DMN/20230324/FEEL/">
+    <itemComponent id="_93C9A060-419D-46DE-925E-D9C0A8253F0C" name="Capacity" isCollection="false" typeLanguage="https://www.omg.org/spec/DMN/20230324/FEEL/">
+      <typeRef>number</typeRef>
+    </itemComponent>
+    <itemComponent id="_16275FD6-173D-4B2A-AABC-849DD1B59239" name="Flight Number" isCollection="false" typeLanguage="https://www.omg.org/spec/DMN/20230324/FEEL/">
+      <typeRef>string</typeRef>
+    </itemComponent>
+  </itemDefinition>
+  <businessKnowledgeModel name="has capacity" id="_B0851FBB-28F4-4ED5-8973-1D98AB0E500C">
+    <variable name="has capacity" id="_23EADBD1-90E4-4E3D-B778-E0388C2F7531" typeRef="boolean" />
+    <encapsulatedLogic id="_61F3594A-9AB3-4B35-8376-C7285AB18BF5" kind="FEEL" label="has capacity" typeRef="boolean">
+      <formalParameter id="_0FB5C070-25AC-42FA-BC98-3AB1E98AFE8D" name="flight" typeRef="tFlight" />
+      <formalParameter id="_D71C5780-8346-436D-85A4-50D5A4E66084" name="rebooked list" typeRef="tPassengerTable" />
+      <context id="_3591FC67-8A44-4E38-8A89-79EACFEF3627" typeRef="boolean" label="Return">
+        <contextEntry id="_3BC7BA57-350B-4A32-A70F-D8FEEF94754C">
+          <variable id="_73A97498-7980-4005-88DE-50E961D6CE6B" name="rebooked list per flight" typeRef="tPassengerTable" />
+          <filter id="_D041B957-7EE3-4888-9AB8-629D6CB6B3AF" label="rebooked list per flight">
+            <in id="_46659CCF-3A7F-45D6-83CB-F79EF22DAC93">
+              <literalExpression id="_38FAED08-B9D7-4A96-A3F5-ED86C7B14E22" typeRef="&lt;Undefined&gt;" label="Expression Name">
+                <text>rebooked list</text>
+              </literalExpression>
+            </in>
+            <match id="_37E842D1-05D0-48F4-8B42-C068E3EFC742">
+              <literalExpression id="_D4B86919-2730-4BE5-BB22-01F7244A290E" typeRef="&lt;Undefined&gt;" label="Expression Name">
+                <text>item.Flight Number = flight.Flight Number</text>
+              </literalExpression>
+            </match>
+          </filter>
+        </contextEntry>
+        <contextEntry id="_232BA058-8B0C-49CD-9DAA-80CAE5750ED4">
+          <conditional id="_4B0C53B1-AB7D-42B8-B79C-B2DCF24FA940" label="Return">
+            <if id="_31DAD263-7069-4F84-B5A7-23023A94207F">
+              <literalExpression id="_838AC5EC-F8F3-49BD-A2D8-0D09B72B372E" typeRef="&lt;Undefined&gt;" label="Expression Name">
+                <text>flight.Capacity &gt; count(rebooked list per flight)</text>
```

Comment:
Even when the filter expression returns a empty list, this will return true.

Reply:
Yes, however I tend to say, it is an expected behavior.

If there is flight that has capacity bigger, than count of rebooked flights, even when there is zero rebooked flights, then the flight has enough capacity for such rebooked flights, i.e. passengers.

Maybe you can explain better the issue?

Reply:
I think yes, will need to change `>` to `>=`, however, otherwise it seems to be fine to me

Reply:
yes, that is expected from my point of view. 

to clarify my thinking. this is more aimed to demo `Filter` expression UI than to solve real domain filtering task. I mean I am sure in the real world, airlines wouldn't use such DMN expression as we have here to check if the flight has capacity or not.


however thank you for the feedback, I will add this NOTE to the stored dmn